src/main/scala/beam/agentsim/agents/ridehail/allocation/EVFleetAllocationManager.scala

```diff
@@ -63,8 +63,7 @@ class EVFleetAllocationManager(val rideHailManager: RideHailManager)
               .remainingRangeInM < routingResponses.get
               .map(
                 _.itineraries
-                  .filter(_.tripClassifier == RIDE_HAIL)
-                  .headOption
+                  .find(_.tripClassifier == RIDE_HAIL)
                   .map(_.beamLegs().map(_.travelPath.distanceInM).sum)
```

Comment:
use fold instead of map and gotOrElse

Reply:
disagree is a good choice. Sum is clearer and the code is already too confuse.